src/x/config/hostid/hostid.go

```diff
@@ -83,17 +88,30 @@ type FileConfig struct {
 	Timeout *time.Duration `yaml:"timeout"`
 }
 
+// HostnameConfig contains the info needed to construct a HostnameConfig.
+type HostnameConfig struct {
+	// Format is a custom format to use, using go templates.
+	// i.e. Using the M3DB operator if using no ID based on the disk
+	// you can format it how the operator would expect:
```

Comment:
nit: comment is a bit janky, also should probably mention that `{{.Hostname}}` is the only variable you can really template for?

Reply:
True, maybe might fix later.